File: frontend/src/api/operationTool.ts
// Copyright (c) 2025 kk
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

import apiClient from './client'

// 运维工具接口
export interface OperationTool {
  id: number
  name: string
  description?: string
  category?: string
  icon?: string
  url: string
  order: number
  enabled: boolean
  created_at: string
  updated_at: string
}

// 创建运维工具请求
export interface CreateOperationToolRequest {
  name: string
  description?: string
  category?: string
  icon?: string
  url: string
  order?: number
  enabled?: boolean
}

// 更新运维工具请求
export interface UpdateOperationToolRequest {
  name?: string
  description?: string
  category?: string
  icon?: string
  url?: string
  order?: number
  enabled?: boolean
}

// 列表查询参数
export interface ListOperationToolsParams {
  category?: string
  enabled?: boolean
}

export const operationToolApi = {
  // 获取工具列表
  list: (params?: ListOperationToolsParams) => {
    const queryParams = new URLSearchParams()
    if (params?.category) {
      queryParams.append('category', params.category)
    }
    if (params?.enabled !== undefined) {
      queryParams.append('enabled', params.enabled.toString())
    }
    const query = queryParams.toString()
    return apiClient.get<{ data: OperationTool[] }>(
      `/operation-tools${query ? `?${query}` : ''}`
    )
  },

  // 获取单个工具
  get: (id: number) =>
    apiClient.get<{ data: OperationTool }>(`/operation-tools/${id}`),

  // 创建工具（管理员）
  create: (data: CreateOperationToolRequest) =>
    apiClient.post<{ data: OperationTool }>('/operation-tools', data),

  // 更新工具（管理员）
  update: (id: number, data: UpdateOperationToolRequest) =>
    apiClient.put<{ data: OperationTool }>(`/operation-tools/${id}`, data),

  // 删除工具（管理员）
  delete: (id: number) =>
    apiClient.delete<{ message: string }>(`/operation-tools/${id}`),
}

File: backend/internal/model/operation_tool.go
// Copyright (c) 2025 kk
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

package model

import "time"

// OperationTool 运维工具
type OperationTool struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Name        string    `gorm:"not null;size:255" json:"name"`
	Description string    `gorm:"type:text" json:"description"`
	Category    string    `gorm:"size:100" json:"category"`
	Icon        string    `gorm:"size:255" json:"icon"`
	URL         string    `gorm:"not null;size:512" json:"url"`
	OrderIndex  int       `gorm:"default:0" json:"order"`
	Enabled     bool      `gorm:"default:true" json:"enabled"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// TableName 指定表名
func (OperationTool) TableName() string {
	return "operation_tools"
}
